### ISSUE TITLE
Document the undocumented Kitsune REST APIs on readthedocs

### ISSUE THREAD

**Issue by rtanglao** [mozilla/kitsune#3475]
**tl;dr Assuming we keep the APIs, it would be nice to document the undocumented REST APIs and their limitations  on readthedocs. Would this be a suitable project for a contributor?**

@akatsoulas @ophie200 @havardl I'm unencumbered by DJANGO and API knowledge :-) so please chime in if I've missed anything.

## Full Details:

1. Kitsune has many REST APIs but only the locales and search suggest API are documented on readthedocs: 
https://kitsune.readthedocs.io/en/latest/api.html#locales-api
1. Assuming we keep the APIs (i.e. we don't switch to graphql or other API framework), it would be nice to document the REST APIs and their limitations (like no way to to specify a time period & only 20 questions at a time in the questions API (```api/2/question/```), no way to tell if a question has been deleted, etc)

### Undocumented APIs from https://github.com/mozilla/kitsune/blob/master/kitsune/settings.py : 

```python
CORS_URLS_REGEX = [
   r'^/api/1/gallery/.*$',
   r'^/api/1/kb/.*$',
   r'^/api/1/products/.*',
   r'^/api/1/users/get_token$',
   r'^/api/1/users/test_auth$',
   r'^/api/2/answer/.*$',
   r'^/api/2/pushnotification/.*$',
   r'^/api/2/notification/.*$',
   r'^/api/2/question/.*$',
   r'^/api/2/realtime/.*$',
   r'^/api/2/search/.*$',
   r'^/api/2/user/.*$',
]
```
### Undocumented APIs from https://github.com/mozilla/kitsune/blob/master/kitsune/urls.py : 

```python
    (r'^api/1/kb/', include('kitsune.wiki.urls_api')),
    (r'^api/1/products/', include('kitsune.products.urls_api')),
    (r'^api/1/customercare/', include('kitsune.customercare.urls_api')),
    (r'^api/1/gallery/', include('kitsune.gallery.urls_api')),
    (r'^api/1/users/', include('kitsune.users.urls_api')),

    # v2 APIs
    (r'^api/2/', include('kitsune.notifications.urls_api')),
    (r'^api/2/', include('kitsune.questions.urls_api')),
    (r'^api/2/', include('kitsune.search.urls_api')),
    (r'^api/2/', include('kitsune.community.urls_api')),
    (r'^api/2/', include('kitsune.sumo.urls_api')),

    # These API urls include both v1 and v2 urls.
(r'^api/', include('kitsune.users.urls_api')),
```

**Comment by ophie200** [comment]
I think this would be a great contributor project. SurveyGizmo does a great job of documenting their [API](https://apihelp.surveygizmo.com/help) if you want a template suggestion. I don't think the Kitsune APIs are going anywhere and will probably continue to be improved so having front-end documentation is useful - so we don't have to back out REST APIs from the django code ;)

**Comment by jscher2000** [comment]
I notice on [the Platform Roadmap](https://app.productplan.com/p/UJazNvHDlwzGDylaBQid7AxEhOOnE5yd) there is a bright pink bar titled "Extract the data layer/ API work" that recently started (2019-07-24 - 2019-12-31). 

Does this API work impact the current APIs? Is there a place one could follow what is planned/changing with the APIs?

**Comment by akatsoulas** [comment]
A new meta issue will be created with all the details and the architecture of the new set of APis
